model/jpa/src/main/java/org/keycloak/models/jpa/JpaRealmProvider.java

```diff
@@ -152,6 +152,16 @@ public Stream<RealmModel> getRealmsStream() {
         return getRealms(query);
     }
 
+    @Override
+    public Stream<RealmModel> getRealmsStream(String search) {
+        if (search.trim().isEmpty()) {
+            return getRealmsStream();
+        }
+        TypedQuery<String> query = em.createNamedQuery("getRealmIdsWithNameContaining", String.class);
+        query.setParameter("search", search);
```

Comment:
Is this worth it?

```
query.setParameter("limit", limit);
query.setParameter("offset", offset);

```

Reply:
This PR is closed. If you think there is the need to optimize it, feel free to open an issue and a PR.